website/source/docs/configuration/storage/s3.html.md

```diff
@@ -82,4 +90,62 @@ storage "s3" {
 }
 ```
 
+### S3 Server Side Encryption (AES256)
+
+This example shows using Amazon S3 as a storage backend with AES256 server side
+encryption enabled.
+
+```hcl
+storage "s3" {
+  access_key = "abcd1234"
+  secret_key = "defg5678"
+  bucket     = "my-bucket"
+  sse        = "AES256"
+}
+```
+
+### S3 Server Side Encryption (KMS)
+
+This example shows using Amazon S3 as a storage backend using KMS
+encryption with the default S3 KMS key for the account.
+
+```hcl
+storage "s3" {
+  access_key = "abcd1234"
+  secret_key = "defg5678"
+  bucket     = "my-bucket"
+  sse        = "aws:kms"
+}
+```
+
+### S3 Server Side Encryption (Custom KMS Key)
+
+This example shows using Amazon S3 as a storage backend using KMS
+encryption with a customer managed KMS key.
+
+```hcl
+storage "s3" {
+  access_key = "abcd1234"
+  secret_key = "defg5678"
+  bucket     = "my-bucket"
+  sse        = "aws:kms"
```

Comment:
It seems like there is no use case for using AES256 as the value for SSE (given that you can't also specify a key, and if you did, it'd be sitting in plaintext). Why not juse only have kms_key_id and test whether it's empty (use default) or whether it's something custom?

Reply:
Could definitely do that. It would simplify things and I'm pretty sure anyone trying to add encryption on top of what vault already does is going to use KMS, unless I add the option to specify your own key material for AES encryption. Either way, I think you're right. It makes more sense to specify a key ID and set the SSE option appropriately. AES support could be added with an alternate parameter if anyone wants it.

Reply:
Updated. Definitely cleans the code up a good bit.